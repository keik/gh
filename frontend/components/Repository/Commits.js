// @flow

import * as React from 'react'
import { jsx, css } from '@emotion/core'
import * as React from 'react'
import { connect } from 'react-redux'
import { Link, type Match } from 'react-router-dom'
import type { Dispatch } from 'redux'

import Button from '../common/atoms/Button'
import SegmentedButtonsContainer from '../common/layouts/SegmentedButtonsContainer'
import type { ReducersStateT } from '../../ducks'
import * as CommitsAction from '../../ducks/repository/commits'
import type { CommitT } from '../../../types/gh'

type Props = {
  commits: Array<CommitT>,

  match: $Shape<
    Match<{
      owner: string,
      repo: string
    }>
  >
}
export default connect<_, _, *, _, *, _>(({ commits }: ReducersStateT) => ({
  commits
}))(
  class CommitsContainer extends React.Component<
    Props & { dispatch: Dispatch<*> }
  > {
    async componentDidMount() {
      const {
        dispatch,
        match: {
          params: { owner, repo }
        }
      } = this.props
      dispatch(await CommitsAction.fetch({ owner, repo }))
    }

    render() {
      return <Commits {...this.props} />
    }
  }
)

export const Commits: React.StatelessFunctionalComponent<Props> = ({
  commits,
  match: { params }
}: *) => {
  return (
    <ul
      css={css`
        padding: 0;
        margin: 0;
        list-style: none;
        border-right: 1px solid #eee;
        border-bottom: 1px solid #eee;
        border-left: 1px solid #eee;
      `}
    >
      {commits.map(commit => (
        <li
          css={css`
            display: flex;
            align-items: center;
            justify-content: space-between;
            padding: 8px;
            border-top: 1px solid #eee;
          `}
          key={commit.sha}
        >
          <div
            css={css`
              display: flex;
            `}
          >
            <div>
              <div
                css={css`
                  font-size: 15px;
                  font-weight: bold;
                  color: #333;
                `}
              >
                {commit.commit.message}
              </div>
              <div
                css={css`
                  font-weight: normal;
                  color: #767676;
                `}
              >
                <Link to={`/${commit.commit.author.name}}`}>
                  <img alt="" src="" />
                </Link>
                {commit.commit.author.name} commited on{' '}
                <time-ago datetime={commit.commit.author.date} />
              </div>
            </div>
          </div>
          <div
            css={css`
              display: flex;
            `}
          >
            <SegmentedButtonsContainer>
              <Button small transparent>
                <i className="fa fa-clipboard" />
              </Button>
              <Button
                as={Link}
                // $FlowFixMe
                small="true"
                to={`/${params.owner}/${params.repo}/commit/${commit.sha}`}
                // $FlowFixMe
                transparent="true"
              >
                {commit.sha.substring(0, 7)}
              </Button>
            </SegmentedButtonsContainer>
            <Button
              as={Link}
              // $FlowFixMe
              small="true"
              style={{ marginLeft: '12px' }}
              to={`/${params.owner}/${params.repo}/tree/${commit.sha}`}
              // $FlowFixMe
              transparent="true"
            >
              <i className="fa fa-code" />
            </Button>
          </div>
        </li>
      ))}
    </ul>
  )
}
