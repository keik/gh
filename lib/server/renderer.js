// @flow

/**
 * Return full HTML string with React props
 *
 * @param {string} content content as HTML string
 * @param {object} props React props object for Server-Side Rendering
 * @param {string} entryName name of entrypoint
 */
export function renderFullPage(
  content: string,
  props: any,
  entryName: string,
  bust: number | string
) {
  return `
 <!DOCTYPE html>
 <html>
   <head>
     <meta charset="utf-8">
     <title>gh</title>
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css">
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.7.0/styles/github.min.css">
     <link href="https://fonts.googleapis.com/css?family=Rajdhani" rel="stylesheet" />
     <link rel="stylesheet" href="/style.css?${bust}">
   </head>
   <body>
     <div id="app">${content}</div>
     <script>APP_PROPS = ${JSON.stringify(props)}</script>
     <script src="/${entryName}.js?${bust}"></script>
   </body>
 </html>
 `
}
