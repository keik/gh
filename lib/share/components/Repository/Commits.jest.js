// @flow

import * as React from 'react'
import renderer from 'react-test-renderer'

import { Commits } from './Commits'

test('<Commits /> snapshot', () => {
  const tree = renderer
    .create(
      <Commits
        commits={[
          {
            commit: {
              author: {
                date: '$DATE_1',
                name: '$NAME_1'
              },
              sha: '$SHA_1',
              message: '$MESSAGE_1'
            },
            html_url: '$HTML_URL_1',
            parents: [{ sha: '$PARENT_SHA_1' }],
            sha: '$SHA_1',
            url: '$URL_1'
          },
          {
            commit: {
              author: {
                date: '$DATE_2',
                name: '$NAME_2'
              },
              sha: '$SHA_2',
              message: '$MESSAGE_2'
            },
            html_url: '$HTML_URL_2',
            parents: [{ sha: '$PARENT_SHA_2' }],
            sha: '$SHA_2',
            url: '$URL_2'
          }
        ]}
        params={{ owner: '$OWNER', repo: '$REPO', sha: '$SHA' }}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
