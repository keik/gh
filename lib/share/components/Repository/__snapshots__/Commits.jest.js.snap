// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Commits /> snapshot 1`] = `
<div
  className="InnerContainer-sc-1toru5t-0 eVVvlf"
>
  <ul
    className="Commits__CommitItems-sc-13zuyrn-0 ibvhoh"
  >
    <li
      className="Commits__CommitItem-sc-13zuyrn-1 knrZNw"
    >
      <div
        className="Commits__CommitInfo-sc-13zuyrn-2 frkLaC"
      >
        <div>
          <div
            className="title"
          >
            $MESSAGE_1
          </div>
          <div
            className="meta"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <img
                alt=""
                src=""
              />
            </a>
            $NAME_1
             commited on
             
            <time-ago
              datetime="$DATE_1"
            />
          </div>
        </div>
      </div>
      <div
        className="Commits__CommitNav-sc-13zuyrn-3 hNNWvi"
      >
        <div
          className="SegmentedButtonsContainer-sc-1alpooo-0 cmuWFA"
        >
          <button
            className="Button-sc-1ldg16x-0 dKuqMW"
          >
            <i
              className="fa fa-clipboard"
            />
          </button>
          <a
            className="Button-sc-1ldg16x-0 dKuqMW"
            onClick={[Function]}
            small="true"
            style={Object {}}
            transparent="true"
          >
            $SHA_1
          </a>
        </div>
        <a
          className="Button-sc-1ldg16x-0 dKuqMW"
          onClick={[Function]}
          small="true"
          style={
            Object {
              "marginLeft": "12px",
            }
          }
          transparent="true"
        >
          <i
            className="fa fa-code"
          />
        </a>
      </div>
    </li>
    <li
      className="Commits__CommitItem-sc-13zuyrn-1 knrZNw"
    >
      <div
        className="Commits__CommitInfo-sc-13zuyrn-2 frkLaC"
      >
        <div>
          <div
            className="title"
          >
            $MESSAGE_2
          </div>
          <div
            className="meta"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <img
                alt=""
                src=""
              />
            </a>
            $NAME_2
             commited on
             
            <time-ago
              datetime="$DATE_2"
            />
          </div>
        </div>
      </div>
      <div
        className="Commits__CommitNav-sc-13zuyrn-3 hNNWvi"
      >
        <div
          className="SegmentedButtonsContainer-sc-1alpooo-0 cmuWFA"
        >
          <button
            className="Button-sc-1ldg16x-0 dKuqMW"
          >
            <i
              className="fa fa-clipboard"
            />
          </button>
          <a
            className="Button-sc-1ldg16x-0 dKuqMW"
            onClick={[Function]}
            small="true"
            style={Object {}}
            transparent="true"
          >
            $SHA_2
          </a>
        </div>
        <a
          className="Button-sc-1ldg16x-0 dKuqMW"
          onClick={[Function]}
          small="true"
          style={
            Object {
              "marginLeft": "12px",
            }
          }
          transparent="true"
        >
          <i
            className="fa fa-code"
          />
        </a>
      </div>
    </li>
  </ul>
</div>
`;
